src/staking/index.ts

```diff
@@ -290,47 +151,32 @@ export class Staking {
   };
 
   /**
-   * Create an unbonding transaction for observable staking.
+   * Create an unbonding transaction for staking.
    * 
-   * @param {ObservableStakingParams} stakingParams - The staking parameters for observable staking.
-   * @param {Delegation} delegation - The delegation to unbond.
+   * @param {Transaction} stakingTx - The staking transaction to unbond.
+   * @param {number} stakingOutputIndex - The output index of the staking transaction.
    * 
-   * @returns {Psbt} - The unsigned unbonding transaction
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
    * 
-   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   * @throws {StakingError} - If the transaction cannot be built
    */
-  public createUnbondingTransaction = (
-    stakingParams: StakingParams,
-    delegation: Delegation,
-  ) : PsbtTransactionResult => {
-    this.validateParams(stakingParams);
-    this.validateDelegationInputs(
-      delegation, stakingParams, this.stakerInfo,
-    );
-    const { 
-      stakingTx,
-      stakingOutputIndex,
-      timelock,
-      finalityProviderPkNoCoordHex,
-      stakerPkNoCoordHex
-    } = delegation;
+  public createUnbondingTransaction(
+    stakingTx: Transaction,
+    stakingOutputIndex: number,
```

Comment:
in theory you could do that, thought isn't it simpler to use `stakingOutputIndex` if you already have it ? I think we already store it in Babylon node database and return it in queries to avoid such re-creation.